src/sec_adapter_soc_provisioning.c

```diff
@@ -0,0 +1,879 @@
+
+#define APPLE_MFI_BASE_KEY_PATH              BASE_PATH"/0391000003910001.key"
+#define APPLE_MFI_PROVISIONING_OBJECT_PATH   BASE_PATH"/0691000006910001.key"
+
+#define APPLE_FAIRPLAY_SECRET_PATH           BASE_PATH"/0691000006910002.bin"
+
+
+static const ObjectIdTypeMapping object_id_type_map[] = {
+    // Widevine
+    { WIDEVINE_PRIVATE_KEY, OBJECT_TYPE_KEY },
+    { WIDEVINE_CERT_BUNDLE, OBJECT_TYPE_BUNDLE },
+
+    // PlayReady
+    { PLAYREADY_2K_PRIVATE_KEY, OBJECT_TYPE_KEY },
+    { PLAYREADY_3K_PRIVATE_KEY, OBJECT_TYPE_KEY },
+    { PLAYREADY_2K_CERTIFICATE, OBJECT_TYPE_CERT },
+    { PLAYREADY_3K_CERTIFICATE, OBJECT_TYPE_CERT },
+
+    // Netflix
+    { PROV_NETFLIX_KDE, OBJECT_TYPE_KEY },
+    { PROV_NETFLIX_KDH, OBJECT_TYPE_KEY },
+    { PROV_NETFLIX_KDW, OBJECT_TYPE_KEY },
+    { PROV_NETFLIX_ESN, OBJECT_TYPE_BUNDLE },
+    { PROV_NETFLIX_DRMBIN, OBJECT_TYPE_BUNDLE },
+
+    // Apple MFi
+    { MFI_BASE_KEY, OBJECT_TYPE_KEY },
+    { MFI_PROVISIONING_OBJECT, OBJECT_TYPE_BUNDLE },
+
+    // Apple FairPlay
+    { FAIRPLAY_SECRET, OBJECT_TYPE_KEY }
+
+};
+
+static SEC_OBJECTID Soc_object_ids[] = {
+    // Widevine
+    WIDEVINE_PRIVATE_KEY,
+    WIDEVINE_CERT_BUNDLE,
+
+    // PlayReady
+    PLAYREADY_2K_PRIVATE_KEY,
+    PLAYREADY_3K_PRIVATE_KEY,
+    PLAYREADY_2K_CERTIFICATE,
+    PLAYREADY_3K_CERTIFICATE,
+
+    // Netflix
+    PROV_NETFLIX_KDE,
+    PROV_NETFLIX_KDH,
+    PROV_NETFLIX_KDW,
+    PROV_NETFLIX_ESN,
+    PROV_NETFLIX_DRMBIN,
+
+    // Apple MFi
+    MFI_BASE_KEY,
+    MFI_PROVISIONING_OBJECT,
+
+    // Apple FairPlay
+    FAIRPLAY_SECRET
+ 
+};
+
+struct Sec_KeyHandle_struct {
+    Sec_ProcessorHandle* processorHandle;
+    Sec_KeyType key_type;
+    Sec_Key key;
+};
+
+
+uint8_t* readBlob(const char* filename, size_t* outSize);
+bool readWidevineData(WidevineOemProvisioning** wvProvision, const char* file_private_key, const char* file_oem_cert);
+bool readPlayReadyData(PlayReadyProvisioning** prProvision,
+                         const char* file_private_key,
+                         const char* file_oem_cert, unsigned int model_type);
+bool readAppleFairPlayData(AppleFairPlayProvisioning** fairPlayProvision,
+                             const char* file_fairplay_secret_name);
+bool readAppleMfiData(AppleMfiProvisioning** mfiProvision,
+                        const char* file_base_key_name,
+                        const char* file_provisioning_object_name);
+bool readNetflixData(NetflixProvisioning** nflxProvision,
+                     const char* file_esn_name,
+                     const char* file_encryption_key_name,
+                     const char* file_hmac_key_name,
+                     const char* file_wrapping_key_name);
+Sec_Result SecSocProv_Soc_Vendor_Specific(SEC_OBJECTID object_id);
+ObjectType get_object_type(SEC_OBJECTID object_id);
+Sec_Result store_raw_data(Sec_ProcessorHandle* processorHandle, Sec_StorageLoc location, SEC_OBJECTID object_id,
```

Comment:
Let's find a name for this function that aids readability. I think the differentiating factor is not in how we write the file but rather whether the keys are exported in node locking format or not.